opal/mca/btl/usnic/btl_usnic_component.c

```diff
@@ -83,7 +83,7 @@
 #define OPAL_BTL_USNIC_NUM_COMPLETIONS 500
 
 /* MPI_THREAD_MULTIPLE_SUPPORT */
-opal_recursive_mutex_t btl_usnic_lock = OPAL_RECURSIVE_MUTEX_STATIC_INIT;
+opal_recursive_mutex_t btl_usnic_lock;  /* recursive mutexes must be dynamically initialized */
```

Comment:
I'm not sure I understand this change -- does it need something like what you did in ompi/instance/instance.c?

Reply:
The lock is initialized during component initialization: https://github.com/open-mpi/ompi/blob/787dd7bf65b3e8f5ebb0a64b455415e549364abe/opal/mca/btl/usnic/btl_usnic_component.c#L590

Reply:
Ah, ok.  👍

Reply:
Correct. All components have a natural constructor, aka. their _init function, and no other function from the component can be used before it successfully returns. usnic BTL is correctly constructing the mutex in the _init, the static initialization was superfluous.

Unfortunately the same is not true for`instance`, I did not see an obvious initialization point inside the `instance.c` file. I imagine we could do something in one of the MPI initialization functions, but then we would need to either expose the recursive mutex, or expose an initializer for the `instances`.